src/meadowgrid/deployed_function.py

```diff
@@ -171,20 +216,18 @@ class MeadowGridVersionedDeployedRunnable:
     TODO this interface is very incomplete
     """
 
-    code_deployment: Union[CodeDeployment, VersionedCodeDeployment, None]
+    code_deployment: Union[CodeDeployment, VersionedCodeDeployment]
```

Comment:
Not exactly - I took it out because in practice when that class is fully instantiated the value is never `None` - if it is it gets a default value `ServerAvailableFolder()`.  Usages of `runnable.interpreter_deployment` don't expect (and reasonably can't deal with) `None`. E.g. coordinator_client.py:246. 

So it seems like we're dealing with the annoying problem that we want some field of a dataclass to have a default value, and it can be passed as an optional argument from a method as well (i.e. `grid_map`), and clearly we don't want to have to write `code_deployment = ServerAvailableFolder()` everywhere, not to mention that for function arguments that mutable object would only be instantiated once instead of on every call (although it probably doesn't matter for this particular thing, it's still annoying and a potential trap later).

 Hm. Open to suggestions. Not using a dataclass for it so we can type the ctor arguments separately from the fields? Use `InitVar`?

Reply:
For now will just add a note to it so I can get this merged